pkg/target/kusttarget_configplugin.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package target
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/pkg/image"
+	"sigs.k8s.io/kustomize/pkg/plugins"
+	"sigs.k8s.io/kustomize/pkg/transformers"
+	"sigs.k8s.io/kustomize/pkg/transformers/config"
+	"sigs.k8s.io/kustomize/pkg/types"
+	"sigs.k8s.io/kustomize/plugin/builtingen"
+	"sigs.k8s.io/yaml"
+)
+
+// Functions dedicated to configuring the builtin
+// transformer and generator plugins using config data
+// read from a kustomization file.
+//
+// Non-builtin plugins will get their configuration
+// from their own dedicated structs and yaml files.
+//
+// There are some loops in the functions below because
+// the kustomization file would, say, allow one to
+// request multiple secrets be made, or run multiple
+// image tag transforms, so we need to run the plugins
+// N times (plugins are easier to write, configure and
+// test if they do just one thing).
+
+func (kt *KustTarget) configureBuiltinGenerators() (
+	[]transformers.Generator, error) {
+	var result []transformers.Generator
+	for _, args := range kt.kustomization.SecretGenerator {
```

Comment:
Maybe split this into different functions `configureBuiltinSecretGenerator`, `configureBuiltinConfigMapGenerator`.

Do you plan to move the resource loading logic to be a builtin generator?

Reply:
done.   Loader remains an argument to plugins, not a plugin itself.